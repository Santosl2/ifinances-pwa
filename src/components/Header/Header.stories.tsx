import { Provider } from "react-redux";

import { makeStore } from "@/store";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import { Header } from "./Header";

export default {
  title: "Components/Header",
  component: Header,
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => (
  <Provider store={makeStore}>
    <Header />
  </Provider>
);

export const Default = Template.bind({});
