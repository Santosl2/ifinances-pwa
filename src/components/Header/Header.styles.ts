import { transparentize } from "polished";
import styled from "styled-components";

export const HeaderWrapper = styled.header`
  background: linear-gradient(180deg, #363f5f 0%, #202538 100%);
`;

export const HeaderContent = styled.div`
  max-width: 1120px;
  margin: 0 auto;
  padding: 2rem 1rem 12rem;
  display: flex;
  align-items: center;
  justify-content: space-between;

  img {
    width: 220px;
  }

  button {
    padding: 0 1.7rem;
    height: 2.5rem;
    background: var(--blue-600);

    &:hover {
      background: ${transparentize(0.3, "#22293f")};
    }
  }

  @media screen and (max-width: 768px) {
    img {
      width: 170px;
    }
  }
`;
