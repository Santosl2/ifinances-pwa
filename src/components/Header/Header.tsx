import { useCallback } from "react";
import { useDispatch } from "react-redux";

import { Button } from "@/components/Form";
import { closeModal } from "@/store/modal/ModalReducers";

import { HeaderContent, HeaderWrapper } from "./Header.styles";

export function Header(): JSX.Element {
  const dispatch = useDispatch();

  const handleOpenModal = useCallback(
    () => dispatch(closeModal({ isOpen: true })),
    []
  );
  return (
    <HeaderWrapper>
      <HeaderContent>
        <img src="/iFinances.svg" alt="iFinances Logo" />
        <Button onClick={handleOpenModal}>Nova transação</Button>
      </HeaderContent>
    </HeaderWrapper>
  );
}
