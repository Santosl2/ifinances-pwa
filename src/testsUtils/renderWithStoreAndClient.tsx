/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/prop-types */
import { ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { Provider } from "react-redux";
import { ToastContainer } from "react-toastify";

import { RootState } from "@/store";
import { combinedReducer } from "@/store/reducers";
import { configureStore } from "@reduxjs/toolkit";
import { render } from "@testing-library/react";

export const queryClientTest = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const testStore = (state: Partial<RootState>) => {
  return configureStore({
    reducer: combinedReducer,
    preloadedState: state,
  });
};

export const renderWithStoreAndClient = (
  component: any,
  initialState: any = {}
) => {
  function Wrapper({ children }: { children: ReactNode }) {
    return (
      <Provider store={testStore(initialState)}>
        <QueryClientProvider client={queryClientTest}>
          {children}
        </QueryClientProvider>
        <ToastContainer />
      </Provider>
    );
  }
  return render(component, { wrapper: Wrapper });
};
