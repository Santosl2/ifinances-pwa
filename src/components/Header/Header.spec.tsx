import * as React from "react";

import { renderWithStoreAndClient } from "@/testsUtils/renderWithStoreAndClient";
import { screen } from "@testing-library/react";

import { Header } from "./Header";

const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  useDispatch: () => mockDispatch,
}));

describe("Header Test", () => {
  beforeEach(() => {
    // useDispatchMock.mockImplementation(() => () => {});
  });

  // afterEach(() => {
  //   useDispatchMock.mockClear();
  // });

  it("should render Header", () => {
    renderWithStoreAndClient(<Header />);

    expect(screen.getByText("Nova transação")).toBeInTheDocument();
    expect(screen.getByAltText("iFinances Logo")).toBeInTheDocument();
  });

  it("should be match snapshot", () => {
    const { container } = renderWithStoreAndClient(<Header />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it.todo("should be open modal");
  // renderWithStore(<Header />);
  // const useDispatchSpy = jest.spyOn(redux, "useDispatch");
  // const mockDispatchFn = jest.fn();
  // useDispatchSpy.mockReturnValue(mockDispatchFn);
  // const btnClick = screen.getByText("Nova transação");
  // fireEvent(btnClick, new MouseEvent("click", { bubbles: true }));
  // expect(mockDispatchFn).toHaveBeenCalled();
});
