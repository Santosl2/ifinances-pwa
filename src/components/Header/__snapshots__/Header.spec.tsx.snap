// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Test should be match snapshot 1`] = `
<header
  class="sc-jSMfEi eauWbA"
>
  <div
    class="sc-gKXOVf gAOQsw"
  >
    <img
      alt="iFinances Logo"
      src="/iFinances.svg"
    />
    <button
      class="sc-hKMtZM cNKgWr"
    >
      Nova transação
    </button>
  </div>
</header>
`;
